src/ctia/flows/hooks/kafka.clj

```diff
@@ -0,0 +1,42 @@
+(ns ctia.flows.hooks.kafka
+  (:require [clojure.tools.logging :as log]
+            [clojure.string :as str]
+            [ctia.flows.hook-protocol :refer [Hook]]
+            [ctia.properties :refer [properties]]
+            [cheshire.core :as json]
+            [franzy.clients.producer.defaults :as pd]
+            [franzy.clients.producer.protocols :as kafka]
+            [franzy.clients.producer.client :as producer]
+            [franzy.serialization.serializers :as serializers]
+            [clients.core :as c]))
+
+(def production-key "data")
+
+(defrecord KafkaPublisher [producer topic partition opts]
+  Hook
+  (init [_]
+    :nothing)
+  (destroy [_]
+    (.close producer))
+  (handle [_ event _]
+    (let [event-json (json/encode event)]
+      (c/retry-exp
+       "send to Kafka"
```

Comment:
The `send-async` function will return immediately once the record has been stored in the buffer. So I don't think that the `retry-exp` is useful here.

Reply:
Good point. It came from having differing ideas at different times, and not merging them correctly.
I don't want to block, since I don't know what this effect this has on the rest of CTIA. But I'm also loathe to fire-and-forget with a `send-async!`.
What I'd like to do here is keep the retry, change to a blocking `send`, and wrap it all in a future. Are you aware of problems with that?

Reply:
No it should work.